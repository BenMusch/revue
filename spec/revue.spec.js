import Component from './component';
const Vue = require('vue');
const Revue = require('../dist/index');

let mocks = {
  value: 'Something Different',
  name: 'Grace Hopper'
}


describe('Revue', () => {

  test('new Revue allows to acces a Vue component', () => {
    let rv = new Revue(Component)
    expect(rv._component).toEqual(Component)
    expect(typeof rv._mounted).toBe('object')
  })

  test('instantiating with mockData will inject it as propsData', () => {
    let rv = new Revue(Component, {name: mocks.name})
    expect(rv._mounted.$props.name).toBe(mocks.name)
  })

  describe('$', () => {
    let rv;

    beforeEach(() => {
      rv = new Revue(Component)
    })

    test('allows to read values of the Vue instance', () => {
      expect(rv.$.desc).toBe('Default description!')
    })

    test('allows to set values on the Vue instance', () => {
      rv.$.desc = mocks.value
      expect(rv.$.desc).toBe(mocks.value)
    })
  })

  describe('$html', () => {
    let rv;

    beforeEach(() => {
      rv = new Revue(Component, {name: mocks.name})
    })

    test('allows to get the rendered html of the instance', () => {
      expect(rv.$html).toMatchSnapshot()
    })

    test('changed data is reflected in the snapshot', () => {
      rv.$.desc = mocks.value
      rv.$tick(() => {
        expect(rv.$html).toMatchSnapshot()
      })
    })
  })
});
