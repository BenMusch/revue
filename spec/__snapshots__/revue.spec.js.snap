// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Revue $html allows to get the rendered html of the instance 1`] = `
"<div>
  <h1>Welcome Grace Hopper</h1> 
  <p class=\\"description\\">Default description!</p> 
  <button>Transform Name</button> 
  <h2>Share some Love</h2> 
  <span class=\\"love\\"></span> 
  <button>Increase</button>
</div>"
`;

exports[`Revue $html changed data is reflected in the snapshot 1`] = `
"<div>
  <h1>Welcome Grace Hopper</h1> 
  <p class=\\"description\\">Something Different</p> 
  <button>Transform Name</button> 
  <h2>Share some Love</h2> 
  <span class=\\"love\\"></span> 
  <button>Increase</button>
</div>"
`;
